src/main/kotlin/calculator/ui/CalculateHandler.kt

```diff
@@ -0,0 +1,19 @@
+package calculator.ui
+
+import calculator.application.Calculator
+import calculator.domain.Expression
+
+class CalculateHandler(
+    private val calculator: Calculator = Calculator()
+) {
+    fun handle(input: String?): Double {
+        validateInput(input)
+        return calculator.calculate(Expression.of(input!!))
+    }
+
+    private fun validateInput(input: String?) {
+        if (input.isNullOrEmpty() || input.isBlank()) {
+            throw IllegalArgumentException("입력받은 문자열이 null이거나 공백입니다")
+        }
+    }
```

Comment:
```suggestion
    fun handle(input: String?): Double {
        if (input.isNullOrEmpty() || input.isBlank()) {
            throw IllegalArgumentException("입력받은 문자열이 null이거나 공백입니다")
        }
        return calculator.calculate(Expression.of(input))
    }
```
이렇게 하나의 함수로 나타내면 `String?#isNullOrEmpty()` 이후에 스마트 캐스팅이 되기 때문에 non-null assertion 연산자(!!)를 피할 수 있습니다. 
메서드로 빼는 장점과 !!을 없앨 수 있는 것 중에 어떤 걸 선택할지 항상 고민되는 것 같습니다. 이 경우는 validation이 그렇게 많지 않아서 한 메서드로 둬도 괜찮지 않나 싶어요. 
참고로, non-null assertion 연산자를 `!!`처럼 불편한 모양으로 정한 건, 사용할 때 한 번 더 생각해보고 불편하게 사용하라는 코틀린 개발진들의 의도가 들어가있다고 합니다 😄  최대한 사용을 지양하는 게 좋을 것 같아요.

Reply:
어쩐지 코드에 `!!`이 들어가 있으면 왠지모르게 불편했는데 다 의도된 것이였네요🥲